pallets/liquid-staking/src/types.rs

```diff
@@ -52,4 +52,46 @@ impl<Balance: BalanceT + FixedPointOperand> MatchingLedger<Balance> {
             Ok((amount - unbonding_amount, unbonding_amount, Zero::zero()))
         }
     }
+
+    pub fn new<T: ExchangeRateProvider>(
+        bond_amount: Balance,
+        rebond_amount: Balance,
+        unbond_amount: Balance,
+    ) -> Result<Self, DispatchError> {
+        let total_stake_amount = bond_amount
+            .checked_add(&rebond_amount)
+            .ok_or(ArithmeticError::Overflow)?;
+        let total_unstake_amount = T::get_exchange_rate()
+            .reciprocal()
+            .and_then(|r| r.checked_mul_int(unbond_amount))
+            .ok_or(ArithmeticError::Overflow)?;
+        Ok(MatchingLedger {
+            total_stake_amount,
+            total_unstake_amount,
+        })
+    }
+}
+
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub enum XcmRequest<T: Config> {
+    Bond {
+        amount: BalanceOf<T>, // KSM amount
```

Comment:
this comment is confusing as it can be DOT as well, I think `liquid_` already means everything